src/main/java/org/terracotta/statistics/derived/latency/DefaultLatencyHistogramStatistic.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * All content copyright Terracotta, Inc., unless otherwise indicated.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.statistics.derived.latency;
+
+import org.terracotta.statistics.Time;
+import org.terracotta.statistics.derived.histogram.BarSplittingBiasedHistogram;
+import org.terracotta.statistics.derived.histogram.Histogram;
+import org.terracotta.statistics.observer.ChainedEventObserver;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.LongSupplier;
+
+import static java.lang.Math.nextDown;
+
+/**
+ * @author Mathieu Carbou
+ */
+public class DefaultLatencyHistogramStatistic implements LatencyHistogramStatistic, ChainedEventObserver {
+
+  private final BarSplittingBiasedHistogram histogram;
+  private final LongSupplier timeSupplier;
+  private final long pruningDelay;
+
+  // histogram querying without any expire call, except one at the start
+  private final Query multiQuery = new Query() {
+    @Override
+    public Long minimum() {
+      return nullOrVal(histogram.getMinimum());
+    }
+
+    @Override
+    public Long maximum() {
+      return nullOrVal(histogram.getMaximum());
+    }
+
+    @Override
+    public long count() {
+      return (long) histogram.getBuckets().stream().mapToDouble(org.terracotta.statistics.derived.histogram.Histogram.Bucket::count).sum();
```

Comment:
Do you call this very often?  The actual histogram implementation knows it's size without having to sum across the bars.  We should expose this on the interface if you need it.

Reply:
oh yes it would be nice because I am calling it often in the simulations. But I won't in Ehcache. But just because it is in the api, we could make it better!

Reply:
I have exposed:

```
  @Override
  public long size() {
    return size;
  }
```

this is the right one ?